Code/Framework/AzCore/AzCore/Math/Uuid.inl

```diff
@@ -17,19 +17,141 @@
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max(),
+                                                                          AZStd::numeric_limits<AZStd::byte>::max() });
```

Comment:
1. This can be populated using a function
2. It would need to account for bytes files not just bytes in the range of 0-127. The Uuid::CreateData needs to work byte value in the range of 0-255
3. There is a tradeoff with the full lookup table creating a 256 byte increase per translation unit where Uuid.h is included in. This lookup table needs to be `inline` and not `static` so there is only one copy at link time

Here is code to fill the lookup table using a function
We have [ranges::iota](https://en.cppreference.com/w/cpp/algorithm/ranges/iota) available in our [ranges_algorithms.h](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/std/ranges/ranges_algorithm.h#L2348-L2391) header which is used as part of the implementation.

```c++
// Lambda expression populates the CharToHexDigit lookup table at compile time(it can't be invoked at runtime)
inline constexpr auto CharToHexDigit = []() constexpr
{
    AZStd::array<AZStd::byte, 256> charToHexTable{};
    AZStd::fill(charToHexTable.begin(), charToHexTable.end(), AZStd::numeric_limits<AZStd::byte>::max());
     // fill characters '0' - '9' with the byte range of 0 - 9
    AZStd::ranges::iota(charToHexTable.begin() + '0', charToHexTable.begin() + '9' + 1, 0);
     // fill characters 'A' - 'F' with the byte range of 10-15(for hex digits)
    AZStd::ranges::iota(charToHexTable.begin() + 'A', charToHexTable.begin() + 'G', 10);
     // fill characters 'a' - 'f' with the byte range of 10-15(for hex digits)
    AZStd::ranges::iota(charToHexTable.begin() + 'a', charToHexTable.begin() + 'g', 10);

    return charToHexTable;
};
```

Reply:
iota doesn't compile for AZStd::byte. It looks like byte doesn't overload operator ++, so that's at least part of the problem.

But using the lambda with fill and then assigning the rest of the values still simplifies it significantly.

Reply:
We can just use a for loop then or `AZStd::ranges::for_each_n`.

```
// Lambda expression populates the CharToHexDigit lookup table at compile time(it can't be invoked at runtime)
// It is invoked immediately
inline constexpr auto CharToHexDigit = []() constexpr
{
    AZStd::array<AZStd::byte, 256> charToHexTable{};
    AZStd::fill(charToHexTable.begin(), charToHexTable.end(), AZStd::numeric_limits<AZStd::byte>::max());
     // fill characters '0' - '9' with the byte range of 0 - 9
    AZStd::ranges::for_each_n(charToHexTable.begin() + '0', 10, [], [initialValue = 0](AZStd::byte& valueRef) mutable { valueRef = initialValue++; });
     // fill characters 'A' - 'F' with the byte range of 10-15(for hex digits)
    AZStd::ranges::for_each_n(charToHexTable.begin() + 'A', [], [initialValue = 10](AZStd::byte& valueRef) mutable { valueRef = initialValue++; });
     // fill characters 'a' - 'f' with the byte range of 10-15(for hex digits)
AZStd::ranges::for_each_n(charToHexTable.begin() + 'a', [], [initialValue = 10](AZStd::byte& valueRef) mutable { valueRef = initialValue++; });

    return charToHexTable;
}();
```

Reply:
At that point, it seems more complicated/difficult to parse than just doing a simple
```
    // Lambda expression populates the CharToHexDigit lookup table at compile time(it can't be invoked at runtime)
    // It is invoked immediately
    inline constexpr auto CharToHexDigit = []() constexpr
    {
        AZStd::array<AZStd::byte, 256> charToHexTable{};
        AZStd::fill(charToHexTable.begin(), charToHexTable.end(), AZStd::numeric_limits<AZStd::byte>::max());
        charToHexTable['0'] = AZStd::byte(0);
        charToHexTable['1'] = AZStd::byte(1);
        charToHexTable['2'] = AZStd::byte(2);
        charToHexTable['3'] = AZStd::byte(3);
        charToHexTable['4'] = AZStd::byte(4);
        charToHexTable['5'] = AZStd::byte(5);
        charToHexTable['6'] = AZStd::byte(6);
        charToHexTable['7'] = AZStd::byte(7);
        charToHexTable['8'] = AZStd::byte(8);
        charToHexTable['9'] = AZStd::byte(9);
        charToHexTable['A'] = AZStd::byte(10);
        charToHexTable['B'] = AZStd::byte(11);
        charToHexTable['C'] = AZStd::byte(12);
        charToHexTable['D'] = AZStd::byte(13);
        charToHexTable['E'] = AZStd::byte(14);
        charToHexTable['F'] = AZStd::byte(15);
        charToHexTable['a'] = AZStd::byte(10);
        charToHexTable['b'] = AZStd::byte(11);
        charToHexTable['c'] = AZStd::byte(12);
        charToHexTable['d'] = AZStd::byte(13);
        charToHexTable['e'] = AZStd::byte(14);
        charToHexTable['f'] = AZStd::byte(15);
        return charToHexTable;
    }
    ();
```

Reply:
Then why not just a regular for loop then.

Reply:
👍